x-pack/plugins/security_solution/public/management/pages/trusted_apps/state/items_page.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export interface PageInfo {
+  index: number;
+  size: number;
+}
+
+export interface ItemsPage<T> {
```

Comment:
Kinda yes but I agree it might be premature generalisation at this point. Will review the pattern later, for now I merged it to the main state and removed generic params.